src/1-theory/10-async.adoc

```diff
@@ -0,0 +1,407 @@
+
+=== Short Polling
+
+The first approach that often gets mentioned is short polling, which is the
+client-server equivalent of a child in the backseat of the car asking "are we
+there yet? are we there yet? are we there yet?". The client is making the same
+GET request to the same resource every few seconds, waiting for some sort of
+change in the answer.
+
+[quote,sp3c1,https://stackoverflow.com/questions/4642598/short-polling-vs-long-polling-for-real-time-web-applications]
+----
+00:00:00 C-> Is the cake ready?
+00:00:01 S-> No, wait.
+00:00:01 C-> Is the cake ready?
+00:00:02 S-> No, wait.
+00:00:02 C-> Is the cake ready?
+00:00:03 S-> Yeah. Have some lad.
+00:00:03 C-> Is the other cake ready? ...
+----
+
+The only benefit of short polling is that it is _very_ easy to implement. The
+client can just write a while loop that makes a request, checks for a response,
+then sleeps for a few seconds before making another request.
+
+The main downside with short polling is that it's a very chatty approach, which
+makes it awful to scale. The more clients you have making these requests the
+more load your API has to handle, and it can grow exponentially. Conditional
+network caching can help a bit here, but even then it sucks the battery life on
+mobile devices.
+
+Generally short polling is rubbish, and you should try to avoid it.
+
+=== Long Polling
+
+A slight tweak on the concept of short polling, long polling maintains an open
+connection with the server until the answer comes in.
+
+[quote,sp3c1,https://stackoverflow.com/questions/4642598/short-polling-vs-long-polling-for-real-time-web-applications]
+----
+00:00:00 C-> Is the cake ready?
+00:00:03 S-> Yeah. Have some lad.
+00:00:03 C-> Is the other cake ready?
+----
+
+See here that instead of answering immediately, the server waited 3 seconds
+before responding with a single update.
+
+Generally folks recommend long polling when the client is only waiting for one
+answer, so if they are ordering a single cake and it might be ready soon, then
+great, let that thread focus on replying about the one cake, and when that cake
+is doing the connection is closed. If the client wants multiple status updates,
+or is interested in multiple cakes, this might not be the way to go.
+
+Personal experiences tell me this sort of blocking activity can lead to quite a
+mess when you have more than a smattering of users. Blocking an entire web
+thread for 3 seconds can certainly be a problem when you have a lot of clients
+asking about a lot of cakes.
+
+For example, in Rails land this is a pretty common server config:
+
+[source,ruby]
+----
+workers Integer(ENV['WEB_CONCURRENCY'] || 2)
+threads_count = Integer(ENV['RAILS_MAX_THREADS'] || 5)
+threads threads_count, threads_count
+
+rackup      DefaultRackup
+port        ENV['PORT']     || 3000
+environment ENV['RACK_ENV'] || 'development'
+----
+
+If there was only one dyno running (a.k.a instance, pod, whatever) then 2 * 5 =
```

Comment:
Just to play devil's advocate, `pod` doesn't really hurt in there and it might give those operating in the k8s world a better idea. You could argue "instance" is most commonly associated with AWS instances which would also count as a vendor term 🤷‍♀️ :) Chuck in a droplet there as well! 😂

Reply:
Yeah this is hard. I do not know what to do. Maybe I can just say "web server" lol.